tutor/plugins/v0.py

```diff
@@ -246,12 +246,12 @@ class EntrypointPlugin(BasePlugin):
 
     ENTRYPOINT = "tutor.plugin.v0"
 
-    def __init__(self, entrypoint: pkg_resources.EntryPoint) -> None:
-        self.loader: pkg_resources.EntryPoint
+    def __init__(self, entrypoint: EntryPoint) -> None:
+        self.loader: EntryPoint = entrypoint
         super().__init__(entrypoint.name, entrypoint)
 
     def _load_obj(self) -> None:
-        self.obj = self.loader.load()
+        self.obj = self.loader.load()  # type: ignore[no-untyped-call]
```

Comment:
Let's try to avoid `type: ignore`. How can we make keep this line typed? Same for lines below.

Reply:
The issue we faced is that the `load()` and `select()` methods in the `importlib_metadata` package are untyped methods. Any calls we make to them would throw the `no-untyped-call` error.

Reply:
That's because we should use the typed functions. For instance, instead of:

    importlib_metadata.entry_points().select(group="tutor.plugin.v1")

we should write:

    importlib_metadata.entry_points(group="tutor.plugin.v1")

Notice how the [`EntryPoint.select`](https://github.com/python/importlib_metadata/blob/a6038c3f2bddad2a6113deb5358e911071ec71d1/importlib_metadata/__init__.py#L278) method is untyped but the [`entry_points`](https://github.com/python/importlib_metadata/blob/a6038c3f2bddad2a6113deb5358e911071ec71d1/importlib_metadata/__init__.py#L1001) function is typed.

Reply:
Unfortunately we don't have any alternative method for this untyped call in importlib since the  libraries inherently perform dynamic operations, which are challenging to statically type-check because the types of  objects are not known until runtime.

Reply:
To bypass this issue, we should replace `entrypoint.load()` by `importlib.import_module(entrypoint.value)`.

We should also open a PR in the "importlib_metadata" project to type the `load` method as `def load() -> t.Any`. Would you like to open this PR or should I do it?

Reply:
We'll test it out after replacing `entrypoint.load()`.

I'll open the PR as I already had the `select()` method typed as well.